### PR TITLE
Use QuickJS own JSON methods that were recently added.

### PR DESCRIPTION
**Pull request by PetterS**
Simplifies encoding and not using eval for the JSON feels right.

### CODE REVIEW

File: module.c [hunk 1]
```diff
@@ -319,6 +308,22 @@ static PyObject *context_module(ContextData *self, PyObject *args) {
...
+// Evaluates a Python string as JSON and returns the result as a Python object. Will
+// return _quickjs.Object for complex types (other than e.g. str, int).
+static PyObject *context_parse_json(ContextData *self, PyObject *args) {
+	const char *data;
+	if (!PyArg_ParseTuple(args, "s", &data)) {
+		return NULL;
+	}
+	JSValue value;
+	Py_BEGIN_ALLOW_THREADS;
+	value = JS_ParseJSON(self->context, data, strlen(data), "data.json";
```
Comment (simonlindholm3) [inline module.c]: missing ), and maybe pass `context_parse_json` or similar as the filename argument

File: module.c [hunk 2]
```diff
@@ -319,6 +308,22 @@ static PyObject *context_module(ContextData *self, PyObject *args) {
 	return context_eval_internal(self, args, JS_EVAL_TYPE_MODULE);
 }
 
+// _quickjs.Context.parse_json
+//
+// Evaluates a Python string as JSON and returns the result as a Python object. Will
+// return _quickjs.Object for complex types (other than e.g. str, int).
```
Comment (simonlindholm3) [inline module.c]: What happens on parse failure?

File: module.c (hunk 1 above)
Comment (PetterS) [reply module.c]: Done.

File: module.c (hunk 2 above)
Comment (PetterS) [reply module.c]: There will be a Python exception. I have added a test that shows this.

**Pull request merged**